src/main/java/org/wildfly/naming/client/remote/RemoteServerTransport.java

```diff
@@ -83,7 +85,11 @@ public void handleEnd(final Channel channel) {
             }
 
             public void handleMessage(final Channel channel, final MessageInputStream message) {
+
                 try (MessageInputStream mis = message) {
+                    if(!channel.getConnection().getLocalIdentity().implies(RemoteNamingPermission.getInstance())) {
+                        throw Messages.log.noPermission(channel.getConnection().getLocalIdentity().getPrincipal().getName(), RemoteNamingPermission.getInstance());
```

Comment:
This should use writeExceptionResponse() to write the naming exception.  Throwing before reading the inv ID means the response won't be correlated.